test/parallel/test-bootstrap-modules.js

```diff
@@ -98,6 +98,7 @@ expected.beforePreExec = new Set([
   'Internal Binding contextify',
   'NativeModule internal/vm',
   'NativeModule internal/modules/helpers',
+  'NativeModule internal/modules/sync_hooks',
```

Comment:
Perhaps just call this `hooks`, since we plan to remove the async hooks?

```suggestion
  'NativeModule internal/modules/hooks',
```

Reply:
Because there's `internal/modules/esm/hooks`, and I think it's even more confusing to have two files that are named `hooks`, but then it doesn't make sense to add the new hooks there since this is not esm-only. Then again, I don't want to move that file to `internal/modules` in this PR, because that would make backporting a lot harder.

Reply:
How about `customization_hooks` then? It won’t make sense to call this `sync_hooks` once the async hooks are gone.

Reply:
`customization_hooks` SGTM (and better, because hooks can mean too many things..)